packages/react-instantsearch/src/core/utils.js

```diff
@@ -64,14 +64,12 @@ export const defer = f => {
 };
 
 export function removeEmptyKey(obj) {
-  Object.keys(obj).forEach(
-    key =>
-      (obj[key] &&
-        typeof obj[key] === 'object' &&
-        !isEmpty(obj[key]) &&
-        removeEmptyKey(obj[key])) ||
-      obj[key] === undefined ||
-      (isEmpty(obj[key]) && delete obj[key])
-  );
+  each(obj, (v, k) => {
```

Comment:
why not `Object.keys(obj).forEach()`? It's longer to write but more standard and less lodash 😄